src/liballoc/rc.rs

```diff
@@ -1359,7 +1359,14 @@ trait RcBoxPtr<T: ?Sized> {
 
     #[inline]
     fn inc_strong(&self) {
-        self.inner().strong.set(self.strong().checked_add(1).unwrap_or_else(|| unsafe { abort() }));
+        // We want to abort on overflow instead of dropping the value.
+        // The reference count will never be zero when this is called;
+        // nevertheless, we insert an abort here to hint LLVM at
+        // an otherwise missied optimization.
+        if self.strong() == 0 || self.strong() == usize::max_value() {
```

Comment:
Hm, should we instead have unreachable_unchecked for the == 0 case?

Reply:
Confusingly, doing this:
```diff
-        if self.strong() == 0 || self.strong() == usize::max_value() {
+        if self.strong() == 0 {
+            unsafe { unreachable_unchecked(); }
+        }
+        if self.strong() == usize::max_value() {
             unsafe { abort(); }
```
produces the same output as before any changes (`before` in the linked comparison).